blockbuster/abci/abci.go

```diff
@@ -33,10 +34,22 @@ func NewProposalHandler(logger log.Logger, mempool blockbuster.Mempool) *Proposa
 // the default lane will not have a boundary on the number of bytes that can be included in the proposal and
 // will include all valid transactions in the proposal (up to MaxTxBytes).
 func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
-	return func(ctx sdk.Context, req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
+	return func(ctx sdk.Context, req abci.RequestPrepareProposal) (resp abci.ResponsePrepareProposal) {
+		// In the case where there is a panic, we recover here and return an empty proposal.
+		defer func() {
+			if err := recover(); err != nil {
+				h.logger.Error("failed to prepare proposal", "err", err)
+				resp = abci.ResponsePrepareProposal{Txs: make([][]byte, 0)}
```

Comment:
I think this deviates from the SDK's default behavior which just returns `req.Txs`

Reply:
Won't this potentially have liveness implications? If we return whatever tendermint gives baseapp it might potentially be out of order relative to how BB should order transactions. essentially going to have a back and forth ping pong sesh between the proposer and other validators. Returning an empty list of txs at least allows the network to continue producing new blocks.

Reply:
> Won't this potentially have liveness implications? If we return whatever tendermint gives baseapp it might potentially be out of order relative to how BB should order transactions.

Well there is no BB ordering because `PrepareProposal` completely borked. But I do see your point in that `ProcessProposal` _expects_ a particular lane ordering.

I'm thinking that BB should be smart enough to know how to skip lanes. But maybe this is too much work for now and we can keep this as-is.

Reply:
Ill update it to return the tendermint txs for now, ill add logging so that its clear that bb prepareproposal panicked.